testutils/api/client.py

```diff
@@ -94,3 +94,26 @@ def __subst_path_params(self, url, path_params):
 
     def __make_headers(self, headers):
         return dict(self.headers, **headers)
+
+
+def wait_for_port(port=8080, host="localhost", timeout=10.0):
+    """Wait until a port starts accepting TCP connections.
+    Args:
+        port (int): Port number.
+        host (str): Host address on which the port should exist.
+        timeout (float): In seconds. How long to wait before raising errors.
+    Raises:
+        TimeoutError: The port isn't accepting connection after time specified in `timeout`.
+    """
+    start_time = time.perf_counter()
+    while True:
+        try:
+            with socket.create_connection((host, port), timeout=timeout):
+                break
+        except OSError as ex:
+            time.sleep(0.50)
+            if time.perf_counter() - start_time >= timeout:
+                raise TimeoutError(
+                    "Waited too long for the port {} on host {} to start accepting "
+                    "connections.".format(port, host)
+                ) from ex
```

Comment:
`raise from ex` ? Thanks, I learned [something](https://docs.python.org/3/reference/simple_stmts.html#the-raise-statement).